### PR TITLE
chore/Issue-119: Setup project with Stylelint a11y plugin

### LINKED ISSUE #119: setup project with Stylelint a11y plugin
## Summary

Coming out of https://github.com/ProjectEvergreen/www.greenwoodjs.dev/pull/112, we wanted to capture the action item of [configuring Stylelint with an a11y plugin](https://github.com/ProjectEvergreen/www.greenwoodjs.dev/pull/112#discussion_r1813195520) that is v16 compatible.

## Details

> As for the support, it looks like the issue that you have linked above is (now) closed and has https://github.com/developer-stylechain/stylelint-a11y/issues/4#issuecomment-1927321522 of using 2 other libraries instead of the developer-stylechain project.
>
> Styleline v15 -> https://www.npmjs.com/package/@ronilaukkarinen/stylelint-a11y
> Styleline v16 -> https://www.npmjs.com/package/@double-great/stylelint-a11y

### PR DESCRIPTION
**Pull request by DevLab2425**
## Related Issue

Fix #119 

## Summary of Changes

- Include `@double-great/stylelint-a11y` plugin 
- Make changes to CSS files after plugin rules applied
    - Majority of changes were including `:focus` along side `:hover` use

### CODE REVIEW

File: stylelint.config.js [hunk 1]
```diff
@@ -1,3 +1,9 @@
 export default {
   extends: ["stylelint-config-recommended"],
+  plugins: ["@double-great/stylelint-a11y"], // -> no `extends` https://github.com/double-great/stylelint-a11y/issues/65
+  rules: {
+    "a11y/media-prefers-reduced-motion": null,
```
Comment (DevLab2425) [inline stylelint.config.js:5]: When set as `true`, this rule will result in 2 failures:

```
src/components/side-nav/side-nav.module.css
  31:5  ✖  Expected & #indicator is used with @media (prefers-reduced-motion)  a11y/media-prefers-reduced-motion

src/components/table-of-contents/table-of-contents.module.css
  41:5  ✖  Expected & #indicator is used with @media (prefers-reduced-motion)  a11y/media-prefers-reduced-motion

✖ 2 problems (2 errors, 0 warnings)
```

I wasn't sure how much you wanted to refactor those 2 instances to wrap that rule (and others as well?) in a media query to meet the rule standard. Happy to do so, but wanted to get your thoughts first.

File: lint-staged.config.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 export default {
-  "*.js": ["npm run lint:js --"],
+  "*.js": ["npm run lint:js --", "npm run format --"],
```
Comment (thescientist13) [inline lint-staged.config.js:2]: was this accidental?  seems like this was something from the [previous PR](https://github.com/ProjectEvergreen/www.greenwoodjs.dev/pull/112#discussion_r1810734443)?

File: stylelint.config.js (hunk 1 above)
Comment (thescientist13) [reply stylelint.config.js:5]: Yeah, this was probably an oversight on my part, we have been trying to abide by this in other components
https://github.com/ProjectEvergreen/www.greenwoodjs.dev/blob/main/src/components/capabilities/capabilities.css#L47

So we should handle these cases as well please 🙏

File: lint-staged.config.js (hunk 2 above)
Comment (DevLab2425) [reply lint-staged.config.js:2]: I thought so too, but `main` doesn't seem to have it. Not sure where it was lost. 

https://github.com/ProjectEvergreen/www.greenwoodjs.dev/blob/0ba63b5cac589766e4c9d67a229b733076cc85e5/lint-staged.config.js#L2

Comment (thescientist13) [reply lint-staged.config.js:2]: > Hmm... it was removed. Should I revert this change then?

yes please, as it seems to work just as fine when applying the formatting just to the glob pattern.  didn't seem like it needed to be run twice.

Comment (DevLab2425) [reply lint-staged.config.js]: Ok, removed.

**Review by thescientist13 — APPROVED** [review]
Sweet!  Will get this merged up after the weekly meeting.

**Pull request merged**